instrumentation/httpd/CMakeLists.txt

```diff
@@ -0,0 +1,39 @@
+cmake_minimum_required(VERSION 3.12)
+
+project(opentelemetry-httpd)
+
+find_package(opentelemetry-cpp REQUIRED)
+find_package(Threads REQUIRED)
+find_package(protobuf REQUIRED)
+find_package(gRPC REQUIRED)
+find_package(CURL REQUIRED)
+find_package(Thrift REQUIRED)
+
+add_library(otel_httpd_module SHARED
+  src/otel/mod_otel.cpp
+  src/otel/opentelemetry.cpp
+)
+
+target_compile_options(otel_httpd_module
+  PRIVATE -Wall -Wextra -fpermissive
```

Comment:
Imho we should not allow for `-fpermissive` and fix the underlying issues instead

Reply:
Yup. Fixed with `reinterpret_cast`